### PR TITLE
Fix Coverage reporting, improve coverage

### PR DESCRIPTION
**Pull request by joskuijpers**
Thanks to NervousFish, we now have proper coverage. The current coverage:

screen shot 2017-05-23 at 15 57 40

As you see, the GUI needs a lot of love and so does the crypto.

I disabled coverage for our Java8 call adapter: it is a copy from Square Inc (see license), to make it work with streamsupport (android 6).

### CODE REVIEW

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/cryptography/ED25519.java [hunk 1]
```diff
@@ -30,36 +32,36 @@ public static EdDSAPrivateKey generatePrivateKey() {
         return generatePrivateKey(seed);
     }
 
-    public static EdDSAPrivateKey generatePrivateKey(byte[] seed) {
```
Comment (MatthijsKok) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/cryptography/ED25519.java]: Please leave this public, we will use this outside of this package in the future.

File: BankChain/app/src/main/java/sun/misc/Unsafe.java [hunk 2]
```diff
@@ -1,19 +0,0 @@
...
- * imported jack libs or the classpath</code>
- *
- * This is a known issue at Google but they won't fix it because Jack is deprecated
- * in favor of an unreleased Preview of Android Studio. However, the preview
- * gives even more issues.
- */
-
-@SuppressWarnings("unused")
-public class Unsafe {
-}
```
Comment (MatthijsKok) [inline BankChain/app/src/main/java/sun/misc/Unsafe.java:19]: Why was this removed?

**Review by MatthijsKok — APPROVED** [review]
LGTM. Nice abstraction of coverage.gradle

File: BankChain/app/src/main/java/sun/misc/Unsafe.java (hunk 2 above)
Comment (joskuijpers) [reply BankChain/app/src/main/java/sun/misc/Unsafe.java:19]: Because we now can. This was needed due to a bug in jack and we don't use jack anymore 🎉
